app/stac_api/collection_summaries.py

```diff
@@ -0,0 +1,203 @@
+import logging
+
+import numpy as np
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+logger = logging.getLogger(__name__)
+
+
+def float_in(flt, floats, **kwargs):
+    '''
+    This function is needed for comparing floats in order to check if a
+    given float is member of a list of floats.
+    '''
+    return np.any(np.isclose(flt, floats, **kwargs))
+
+
+def update_summaries_on_asset_delete(collection, asset):
+    '''
+    updates the collection's summaries on an asset's deletion or raises
+    errors when this fails.
+    :param collection: collection, for which the summaries probably need an update
+    :param asset: asset thats being deleted
+    For all the given parameters this function checks, if the corresponding
+    parameters of the collection need to be updated. If so, they will be either
+    updated or an error will be raised, if updating fails.
+    '''
+
+    try:
+        asset_qs = asset.__class__.objects.filter(item__collection_id=collection.pk
+                                                 ).exclude(id=asset.id)
+        if bool(asset_qs):
+
+            if not asset_qs.filter(geoadmin_variant=asset.geoadmin_variant).exists():
+                collection.summaries["geoadmin:variant"].remove(asset.geoadmin_variant)
+                collection.summaries["geoadmin:variant"].sort()
+                collection.save()
+
+            if not asset_qs.filter(proj_epsg=asset.proj_epsg).exists():
+                collection.summaries["proj:epsg"].remove(asset.proj_epsg)
+                collection.summaries["proj:epsg"].sort()
+                collection.save()
+
+            if not asset_qs.filter(eo_gsd=asset.eo_gsd).exists():
+                collection.summaries["eo:gsd"].remove(asset.eo_gsd)
+                collection.summaries["eo:gsd"].sort()
+                collection.save()
+
+        else:
+            # asset was the last item in the collection
+            collection.summaries["geoadmin:variant"] = []
+            collection.summaries["proj:epsg"] = []
+            collection.summaries["eo:gsd"] = []
+            collection.save()
+
+    except KeyError as err:
+        logger.error(
+            "Error when updating collection's summaries values due to asset deletion: %s", err
+        )
+        raise ValidationError(_(
+            "Error when updating collection's summaries values due to asset deletion."
+        ))
```

Comment:
Not sure if this is really needed ? We should always have the `geoadmin:variant`, `proj:espg` and `eo:gsd` keys. Did you have a use case where they were missing ? In any case we should not have a ValidationError here, this error is for user input validation.
From my understanding the only use case that could happen, is during initialization where the summaries haven't been set with default values, but that should not be an error, simply a debug log without raising exception.

Reply:
Is the exception happening with the admin page, when creating new Asset or new Collection ?

Reply:
No, I did not have this error somewhere. Just implemented the exception handling here purely for precaution.
I'll remove it and replace with a debug message, just in case